testbench/database.py

```diff
@@ -332,6 +338,22 @@ def delete_object(
             bucket = self.__get_bucket_for_object(bucket_name, context)
             bucket.pop("%s#%d" % (blob.metadata.name, blob.metadata.generation), None)
 
+    def do_update(
```

Comment:
Please rename this to `do_object_update` or something similar.  There are other "things" in the database and it is unclear what this will act upon.